src/reducers/entities/teams.js

```diff
@@ -16,60 +17,127 @@ function currentId(state = '', action) {
 function teams(state = {}, action) {
     switch (action.type) {
     case TeamsTypes.RECEIVED_ALL_TEAMS:
+    case TeamsTypes.RECEIVED_TEAM_LISTINGS:
         return Object.assign({}, state, action.data);
+
+    case TeamsTypes.CREATED_TEAM:
+    case TeamsTypes.UPDATED_TEAM:
+        return {...state, [action.data.id]: action.data};
+
     case UsersTypes.LOGOUT_SUCCESS:
         return {};
+
     default:
         return state;
     }
 }
 
 function myMembers(state = {}, action) {
-    const nextState = {...state};
-
     switch (action.type) {
+    case TeamsTypes.RECEIVED_MY_TEAM_MEMBER: {
+        const member = action.data;
+        return {
+            ...state,
+            [member.team_id]: member
+        };
+    }
+
     case TeamsTypes.RECEIVED_MY_TEAM_MEMBERS: {
+        const nextState = {};
         const members = action.data;
         for (const m of members) {
             nextState[m.team_id] = m;
         }
         return nextState;
     }
+
     case TeamsTypes.LEAVE_TEAM: {
+        const nextState = {...state};
         const data = action.team;
         Reflect.deleteProperty(nextState, data.team_id);
         return nextState;
     }
     case UsersTypes.LOGOUT_SUCCESS:
         return {};
+
     default:
         return state;
     }
 }
 
 function membersInTeam(state = {}, action) {
     switch (action.type) {
-    default:
-        return state;
+    case TeamsTypes.RECEIVED_MEMBERS_IN_TEAM: {
+        const data = action.data;
+        let teamId;
+        let members;
+
+        if (Array.isArray(data)) {
+            teamId = data[0].team_id;
+            members = new Set(state[teamId]);
```

Comment:
but they depend on the teamId which is dependent of the if array, but nvm I'm going to have different constants for this one